backend/src/sms/sms.service.ts

```diff
@@ -16,15 +16,16 @@ export class SmsService {
     async sendVerificationSMS(recipient: Recipient, challenge: Challenge): Promise<boolean> {
         const settings = SMSConfig.getTeliaConfig();
 
-        // check if any error is thrown while obtaining SMS settings during junior registration (to disply different error message)
-        const smsSettingsError = new Error().stack.split("at ")[2].includes("registerJunior");
+        // check stack trace to see if sendVerificationSMS() is called from registerJunior()
+        const checkRegisterJuniorCalls = new Error().stack.split("at ")[2].includes("registerJunior");
 
-        if(!settings && smsSettingsError) {
-            throw new InternalServerErrorException(content.SMSNotAvailableButUserCreated);
-        }
-
-        else if (!settings) {
-            throw new InternalServerErrorException(content.MessengerServiceNotAvailable);
+        if (!settings) {
+            if(checkRegisterJuniorCalls) {
+                throw new InternalServerErrorException(content.SMSNotAvailableButUserCreated);
+            }
+            else {
```

Comment:
This is unnecessary else. I wouldn't have nested these either to a second level, as there's absolutely no need to. I'll merge it anyway now.